README.md

```diff
@@ -704,22 +719,32 @@ player.tech().vhs.xhr.beforeRequest = function(options) {
 
 The global `videojs.Vhs` also exposes an `xhr` property. Adding
 `onRequest`, `onResponse` hooks and/or specifying a `beforeRequest` 
-function that will allow you to intercept the request Object, response 
-data and options for *all* requests in every player on a page. For 
-consistency across browsers the video source should be set at runtime 
+function that will allow you to intercept the request options, xhr 
+Object, error and response data for *all* requests in every player on a page. 
+For consistency across browsers the video source should be set at runtime 
 once the video player is ready.
 
 Example:
 ```javascript
 // Global request callback, will affect every player.
-const globalRequestHook = (request) => {
-  const requestUrl = new URL(request.uri);
+const globalRequestHook = (options) => {
+  const requestUrl = new URL(options.uri);
   requestUrl.searchParams.set('foo', 'bar');
-  request.uri = requestUrl.href;
+  options.uri = requestUrl.href;
 };
 videojs.Vhs.xhr.onRequest(globalRequestHook);
 ```
 
+```javascript
+// Global request callback defining beforeSend function, will affect every player.
+const globalXhrRequestHook = (options) => {
+  options.beforeSend = (xhr) => {
+    xhr.open('GET', 'https://new.uri');
```

Comment:
is this recommended? Since I'd assume `xhr` would try and call `open` after itself?

Reply:
Good point. Should probably do... something else. Maybe an equivalent example of setting a header?

Reply:
Yeah I wouldn't say I'd recommend it 😅 , it does work however. According to MDN it re-initializes the existing request with every subsequent `open` call. Probably not ideal if applied to _every_ request. I'll think up a better example for the docs.

Reply:
Fixed. Used `setRequestHeader` instead.